### PR TITLE
Feature/nicolaswl/965389861 backend setup

### PR DESCRIPTION
**Pull request by nicolaswl**
### Administrative Info

Monday Board ID: 965389861

### Changes

- Restructured repository to have separate frontend and backend folders
- Created a basic Node/Mongooose/Express backend
- Added routes for creating, modifying, and receiving an agency profile
- Created a database schema for the agency profile

### Testing

- Created sample agency profile requests to verify our routes and form validation 

### Confirmation of Change

- All additions can be seen the in the backend directory

### CODE REVIEW

File: backend/models.js [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    },
+
+    /* Retail Rescue Section */
+    retailRescue: {
+        type: Boolean
+    },
+    preparedFoodCapacity: {
+        type: Boolean
+    },
+    capacityWithRRL: {
```
Comment (sblumin) [inline backend/models.js:247]: What is RRL? Is it supposed to be RRD for RR with delivery?

File: backend/routes/agency.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    body('contacts.*.email').trim().isEmail(),
+    body('scheduledNextVisit').isISO8601(),
+    body('dateOfMostRecentAgreement').trim().isDate({ format: 'MM/DD/YYYY' }),
+    body('dateOfInitialPartnership').trim().isDate({ format: 'MM/DD/YYYY' }),
+    body('fileAudit').trim().isDate({ format: 'MM/DD/YYYY' }),
+    body('monitored').trim().isDate({ format: 'MM/DD/YYYY' }),
+    body('foodSafetyCertification').trim().isDate({ format: 'MM/DD/YYYY' }),
+];
+
+router.put('/', validationChain, async (req, res, next) => {
```
Comment (sblumin) [inline backend/routes/agency.js:21]: Can comments be added to these routes to describe what they're for

**Review by sblumin — CHANGES_REQUESTED** [review]
Looks good! I left two comments, please address these and besides that it all looks good to me!

File: backend/routes/agency.js (hunk 2 above)
Comment (NirmalAgnihotri) [reply backend/routes/agency.js:21]: Seconding this comment

File: backend/models.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    },
+    sunday: {
+        type: Boolean,
+        required: true
+    }
+});
+
+const AgencySchema = new Schema({
+    tableContent: {
+        type: TableContentSchema,
```
Comment (NirmalAgnihotri) [inline backend/models.js:86]: maybe add a brief comment here about how this can be used to access the data in the schema above

**Review by NirmalAgnihotri — APPROVED** [review]
Looks great! Left a comment about a minor detail.

**Review by sblumin — APPROVED** [review]
LGTM! Thanks for addressing those changes!

**Comment by wllmwu** [comment]
Was just looking at this to see how Express routing works--in backend/routes/agency.js, is there a reason that scheduledNextVisit gets validated with isISO8601() while the other dates just use isDate()?

**Comment by StephenTan12** [comment]
> Was just looking at this to see how Express routing works--in backend/routes/agency.js, is there a reason that scheduledNextVisit gets validated with isISO8601() while the other dates just use isDate()?

ISO8601 is another version of date which is the format YYYY-MM-DD, but I should have been changed to isDate(). Thanks for catching it tho, William :)

**Comment by sblumin** [comment]
> > Was just looking at this to see how Express routing works--in backend/routes/agency.js, is there a reason that scheduledNextVisit gets validated with isISO8601() while the other dates just use isDate()?
> 
> ISO8601 is another version of date which is the format YYYY-MM-DD, but it should have been changed to isDate(). Fixed it just now. Thanks for catching it William! :)

good catch Will!!!!!

**Review by StephenTan12 — APPROVED** [review]

**Pull request merged**